### PR TITLE
PICARD-1720: Add scripting $slice() with associated tests.

### PR DESCRIPTION
**Pull request by rdswift**
Add a scripting function to extract (slice) a portion of a multi-value variable. An example would be to use `$setmulti(supporting_artists,$slice(%artists%,1))` to get a multi-value variable with all artists except the first.

# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Add new `$slice_multi()` scripting function with associated tests.**:

# Problem

* JIRA ticket: PICARD-1720

# Solution

Added new `$slice()` scripting function with associated tests.

# Action

### CODE REVIEW

File: picard/script.py [hunk 1]
```diff
@@ -1047,10 +1055,12 @@ def func_map_multi(parser,
...
-        multi: The multi-value to be iterated.
-        loop_code: String of script code to be processed on each iteration that yields the new value for
-            the multi-value element.
-        separator: String used to separate the elements in the multi-value.
+        multi: The ScriptVariable/Function that evaluates to a multi-value which
+            is to be iterated.
+        loop_code: The ScriptVariable/Function of script code to be processed on
+            each iteration that yields the new value for the multi-value element.
+        separator: A string or the ScriptVariable/Function that evaluates to the
+            string used to separate the elements in the multi-value.
```
Comment (zas) [inline picard/script.py]: Changes related to documentation of previous functions should go in a separate PR.

Comment (rdswift) [reply picard/script.py]: I backed them out of this PR and will create a new PR (later when the dust settles a little).  :-)

**Comment by phw** [comment]
It probably makes sense to merge #1453 first with the decorator, then adapt this PR to it.

**Comment by zas** [comment]
> It probably makes sense to merge #1453 first with the decorator, then adapt this PR to it.

Yes, should we rename this to `func_slice` (so we have `$slice` script function) ?

**Comment by rdswift** [comment]
I left it as `$slicemulti()` for now, but can easily change it to `$slice()` if that's what you'd prefer.

**Comment by phw** [comment]
@rdswift Yes, let's rename it to `$slice`, I think that reads nicer in scripts.

**Comment by zas** [comment]
> @rdswift Yes, let's rename it to `$slice`, I think that reads nicer in scripts.

Yes, I'm in favor of `$slice` too

**Comment by rdswift** [comment]
TBH, I prefer the shorter name as well.

**Comment by zas** [comment]
@rdswift can you rebase and update PR subject/description ? (getting rid of $slice_multi / $slicemulti occurences)

**Comment by rdswift** [comment]
Updated and rebased.

**Comment by zas** [comment]
> Updated and rebased.
> 
> Update: Oops, just noticed that you already saw that and approved it.

This PR now conflicts with master (because of the other similar PR), can you fix conflicts so we can (finally) merge ? Thanks !
